code/game/turfs/open/floor.dm

```diff
@@ -265,14 +267,15 @@
 			A.update_icon()
 			return TRUE
 		if(RCD_DECONSTRUCT)
-			if(ScrapeAway(flags = CHANGETURF_INHERIT_AIR) == src)
```

Comment:
I have no clue what this check was meant to do, but I don't think it did anything other than always return false and cause the `to_chat` message to never display.

Reply:
I think it's meant to prevent deconstruction from doing anything if the turf is the same as baseturf. Like on lavaland, the baseturf is volcanic floor, if you tried to deconstruct it, it would fail, since the baseturf of volcanic floor is volcanic floor.

Reply:
Also ChangeTurf will return the new turf, so it's checking if the turf changed